src/PDDoc.jl

```diff
@@ -181,3 +182,55 @@ function pdDocGetNamesDict(doc::PDDoc)
     ref = get(catalog, CosName("Names"))
     obj = cosDocGetObject(doc.cosDoc, ref)
 end
+
+"""
+```
+    pdDocGetOutline(doc::PDDoc; depth::Number = Inf, compact::Bool = false, add_index::Bool = false) -> PDOutline
+```
+Given a PDF document provides the document Outline (Table of Contents)
+available in the `Document Catalog` dictionary.
+Returned object is an array of either `PDOutlineItem` or `PDOutline` items.
+Nested `PDOutline` is allways preceded by `PDOutlineItem` and contain its sub-sections.
+First element of `PDOutline` array is allways `PDOutlineItem`.
+
+`PDOutlineItem` represent outline item and is a dictionairy object with following keys:
+- `:Title` - outline displayed title (`CDTextString`), always present
+- `:Level` - nesting level of the item (`Int`), optional, not present in level 1 items or if `:Index` key is present
+- `:Index` - inxedes of the item in Outline structure (`NTuple`), optional, not present if `:Level` key is present.
+             If index is (a,b,c) the item can be referred as outline[a][b][c], where outline is a object returned by this function.
+- `:Expanded` - weather child items should be expanded by the GUI viewer by default (`Bool`), optional
+- `:Style` - the style which should be applied to item by the GUI viewer (`Int`) - refer to PDF specification, optional.
+- `:PageRef` - indirect reference to respective page (`CosIndirectObjectRef`) - use with other API functions eg. `cosDocGetObject`, present if not in compact mode
+- `:PageNo` - absulute number of respective page (`Int`), present if not in compact mode
+- `:PageLabel` - displayable label of respective page (`LabelNumeral` - can be casted to `String`), optional.
+
+Optional, named parameters:
+- `depth` - limits retrieved items to certain nesting level (0 for root chapters), default: no limit
```

Comment:
Outlines are mapped to destinations which use absolute page number in the PDF spec. The page labels need not be mapped to page numbers in this method.

Reply:
There is actually no such map. They simply coexists in the same record. If at all, the map is rather opposite page number -> label.
The reason why there is label is that PDF viewers display it in its outline window and on TOC.
So if you'd e.g. create some text TOC from outline or simply verify what function returns with viewer - you need labels.
PS: Label key is not present when document does not have Labels catalog.